bundles/org.openhab.core.model.core/src/main/java/org/openhab/core/model/core/internal/ModelRepositoryImpl.java

```diff
@@ -98,20 +98,18 @@ public ModelRepositoryImpl(final @Reference SafeEMF safeEmf) {
 
     @Override
     public boolean addOrRefreshModel(String name, final InputStream originalInputStream) {
+        logger.info("Loading model '{}'", name);
         Resource resource = null;
         InputStream inputStream = null;
         try {
-            if (originalInputStream != null) {
-                byte[] bytes = originalInputStream.readAllBytes();
-                String validationResult = validateModel(name, new ByteArrayInputStream(bytes));
-                if (validationResult != null) {
-                    logger.warn("Configuration model '{}' has errors, therefore ignoring it: {}", name,
-                            validationResult);
-                    removeModel(name);
-                    return false;
-                }
-                inputStream = new ByteArrayInputStream(bytes);
+            byte[] bytes = originalInputStream.readAllBytes();
+            String validationResult = validateModel(name, new ByteArrayInputStream(bytes));
+            if (validationResult != null) {
+                logger.warn("Configuration model '{}' has errors, therefore ignoring it: {}", name, validationResult);
+                removeModel(name);
+                return false;
             }
+            inputStream = new ByteArrayInputStream(bytes);
```

Comment:
This is a nice improvement. With absence of these ugly nestings we can use a try-with-resources statement on the `inputStream` declaration and get rid of the `finally` introduced in #1423.